proto/spacemesh/api.proto

```diff
@@ -1,166 +1,70 @@
 syntax = "proto3";
 package spacemesh;
 import "google/protobuf/empty.proto";
+import "spacemesh/types.proto";
 
-message SimpleMessage {
-    string value = 1;
-}
+service SpacemeshService {
 
-message SimpleInt {
-    uint64 value = 1;
-}
+    // A simple test endpoint
+    rpc Echo(SimpleString) returns (SimpleString);
 
-message TxConfirmation {
-    string value = 1;
-    string id = 2;
-}
+    ////////// Node facade
 
-enum TxStatus {
-    UNDEFINED = 0;
-    REJECTED = 1;
-    //INSUFFICIENT_FUNDS = 2;
-    //CONFLICTING = 3;
-    PENDING = 4;
-    //PROCESSING = 5;
-    //APPROVED = 6;
-    CONFIRMED = 7;
-}
+    // Returns the version of the node software as a semver string
+    rpc NodeVersion(google.protobuf.Empty) returns (SimpleString);
 
-message TransactionId {
-    bytes id = 1;
-}
+    // Returns the version of the github tag or branch used to build the node
+    rpc NodeBuild(google.protobuf.Empty) returns (SimpleString);
 
-message Transaction {
-    TransactionId tx_id = 1;
-    AccountId sender = 2;
-    AccountId receiver = 3;
-    uint64 amount = 4;
-    uint64 fee = 5;
-    TxStatus status = 6;
-    uint64 layer_id = 7;
-    uint64 timestamp = 8;
-}
-
-message AccountId {
-    string address = 1;
-}
-
-message Reward {
-    uint64 layer = 1;
-    uint64 total_reward = 2;
-    uint64 layer_reward_estimate = 3;
-    // total_reward - layer_reward_estimate = fees_estimate
-    AccountId recipient = 4;
-}
+    // Returns node status
+    rpc NodeStatus(google.protobuf.Empty) returns (NodeStatus);
 
-message Block {
-    bytes id = 1;
-    repeated Transaction transactions = 2;
-}
+    // Tell the node to start syncing
+    rpc NodeSyncStart(google.protobuf.Empty) returns (google.protobuf.Empty);
 
-message LayerId {
-    bytes id = 1;
-}
+    // Gracefully shut down the node
+    rpc NodeShutdown(google.protobuf.Empty) returns (google.protobuf.Empty);
```

Comment:
What happens here if the node cannot gracefully shutdown for a long time (per @noamnelke's comments)? Does this succeed immediately but the node doesn't actually shutdown right away? Does it fail? Does a sequence of commands need to be sent to gracefully shutdown?

Reply:
You’re referring to the “very graceful shutdown” that I described? If yes, this isn’t really about shutting down... A more fitting name is “graceful retreat from mining”, which may or may not be followed by a node shutdown.

Since the procedure I described would take 1-2 epochs to complete, it makes no sense to keep a request pending this long.

If you’re referring to the regular node shutdown, then I think it makes sense to return a success or failure (e.g. couldn’t dump recovery data to disk), but this would be very difficult to implement.

Reply:
Maybe we need two endpoints, a "shutdown now" endpoint and a "take your time and shutdown gracefully" endpoint :)

Reply:
I don’t think it makes sense to order the node to shut down 2 weeks in advance...

The `StopMining` endpoint should only stop creating ATXs and it should be the caller’s responsibility to monitor and shut down the node when the time is right.

We may also want to make a `shutdown` endpoint that fails if the miner is eligible, unless a `force` flag is also passed.

Reply:
Moved to #10